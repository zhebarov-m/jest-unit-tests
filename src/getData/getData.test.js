const getData = require('./getData')
const axios = require('axios')
const userData = require('./data.json')
jest.mock('axios')

describe('getData', () => {
    let response;
    beforeEach(() => {
        response = {
            data: userData
        }
    })
    test('Корректное значение', async () => {
        axios.get.mockReturnValue(response);
        const data = await getData()
        expect(axios.get).toBeCalledTimes(1);
        expect(data).toEqual(['1', '2', '3'])
        expect(data).toMatchSnapshot()
    })
})