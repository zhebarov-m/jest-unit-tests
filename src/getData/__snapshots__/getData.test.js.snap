// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData Корректное значение 1`] = `
[
  "1",
  "2",
  "3",
]
`;
